shell/machine-config/vmwarevsphere.vue

```diff
@@ -382,12 +381,14 @@ export default {
 
       set(this.value, 'boot2dockerUrl', boot2dockerUrl);
     },
-    vappMode(value) {
+    vappMode(value, oldValue) {
       if (value === VAPP_MODE.AUTO) {
         const defaultVappOptions = getDefaultVappOptions(this.networkNames || []);
 
         return this.updateVappOptions(defaultVappOptions);
-      } else {
+      } else if (value === VAPP_MODE.DISABLED) {
+        this.updateVappOptions(INITIAL_VAPP_OPTIONS);
+      } else if (value === VAPP_MODE.MANUAL && oldValue === VAPP_MODE.AUTO) {
```

Comment:
We only want to switch to the initial values if this was on auto. This prevents the manual values from getting rewritten but also overwrites the auto values so we don't see different values when switching from disabled/auto to manual.